_posts/2017-04-12-managing-technical-debt.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: Managing Technical Debt
+description: "Quite obvious plan to manage a technical debt."
+author: bogdan
+modified: 2014-12-24
+tags: [code review, agile, process, qa]
+categories: [process]
+image:
+    feature: posts/tech_debt/title.jpg
+---
+
+# WHAT IS “TECHNICAL DEBT”?
+
+Martin Fowler [has a great explanation](https://martinfowler.com/bliki/TechnicalDebt.html) of this term and the problem behind it.  Many startups and new projects use quick-and-dirty solutions to achieve their goals as soon as possible, which in some point makes project complicated, slow, hard to change or implement a new features and full of bugs. I want to talk about most usual issues and how manage most effectively from my point of view.
+Project with the big technical debt leading to make developers fear for changes. Any change can have unexpected effect and produce more bugs. And there is only one way to fearless development - it’s automated tests, which such projects usually don’t have. So technical debt is mostly about luck of control of the project quality.  There is other common issues in the project with big TD, but this is the most important one, and, what the worst thing here, very underestimated.
+ <!-- more -->
+ 
+# TL;DR: Plan
+1. **Create automatic backup for application users data.** We don’t want to corrupt user data.
+2. **Start using [commit strategy and workflow](http://nvie.com/posts/a-successful-git-branching-model/)** and make sure that everybody in a team understand it. There should be clear process to publish changes.
```

Comment:
**Start using [commit strategy and workflow](http://nvie.com/posts/a-successful-git-branching-model/)** and make sure that everybody in the team understands it. There should be a clear process for publishing changes.